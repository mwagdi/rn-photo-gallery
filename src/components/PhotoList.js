import React, { Component } from 'react';
import { View } from 'react-native';
import Photo from './Photo';

class PhotoList extends Component{
    componentDidMount(){
        this.props.fetchPhotos();
    }
    render(){
        const { photoIds, photosById, openModal } = this.props;
        return (
            <View>
                {photoIds.map(id => <Photo key={id} photo={photosById[id]} openModal={openModal} />)}
            </View>
        )
    }
}

export default PhotoList;