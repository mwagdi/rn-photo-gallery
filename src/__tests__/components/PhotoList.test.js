import React from 'react';
import renderer from 'react-test-renderer';
import PhotoList from './../../components/PhotoList';

describe('<PhotoList />', () => {
    const photoIds = [1,2,3];
    const photosById = {
        1: {
            id: 1,
            title: "Image 1",
            src: "https://cdn-images-1.medium.com/max/1500/1*dIocy2HvI_BIpziOypf8ig.jpeg",
            description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent et augue ullamcorper, congue augue in, consectetur libero. Vestibulum vel ex.",
            comments: []
        },
        2: {
            id: 2,
            title: "Image 2",
            src: "https://cdn-images-1.medium.com/max/1500/1*dIocy2HvI_BIpziOypf8ig.jpeg",
            description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent et augue ullamcorper, congue augue in, consectetur libero. Vestibulum vel ex.",
            comments: []
        },
        3: {
            id: 3,
            title: "Image 3",
            src: "https://cdn-images-1.medium.com/max/1500/1*dIocy2HvI_BIpziOypf8ig.jpeg",
            description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent et augue ullamcorper, congue augue in, consectetur libero. Vestibulum vel ex.",
            comments: []
        }
    }
    const tree = renderer.create(<PhotoList photoIds={photoIds} photosById={photosById} />).toJSON();
    it('has 3 children', () => {
        expect(tree.children.length).toBe(3);
    });
    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
    });
});