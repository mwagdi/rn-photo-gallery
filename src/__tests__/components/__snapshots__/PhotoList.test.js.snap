// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoList /> renders correctly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Image 1
    </Text>
    <Image
      source={
        Object {
          "uri": "https://cdn-images-1.medium.com/max/1500/1*dIocy2HvI_BIpziOypf8ig.jpeg",
        }
      }
      style={
        Object {
          "height": 100,
          "width": 200,
        }
      }
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Image 2
    </Text>
    <Image
      source={
        Object {
          "uri": "https://cdn-images-1.medium.com/max/1500/1*dIocy2HvI_BIpziOypf8ig.jpeg",
        }
      }
      style={
        Object {
          "height": 100,
          "width": 200,
        }
      }
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Image 3
    </Text>
    <Image
      source={
        Object {
          "uri": "https://cdn-images-1.medium.com/max/1500/1*dIocy2HvI_BIpziOypf8ig.jpeg",
        }
      }
      style={
        Object {
          "height": 100,
          "width": 200,
        }
      }
    />
  </View>
</View>
`;
